dbms/src/Storages/KVStore/tests/gtest_kvstore_fast_add_peer.cpp

```diff
@@ -870,5 +867,64 @@ try
 }
 CATCH
 
+TEST_F(RegionKVStoreTestFAP, FAPWorkerException)
+try
+{
+    CheckpointRegionInfoAndData mock_data = prepareForRestart(FAPTestOpt{});
+    KVStore & kvs = getKVS();
+    RegionPtr kv_region = std::get<1>(mock_data);
+
+    auto & global_context = TiFlashTestEnv::getGlobalContext();
+    auto fap_context = global_context.getSharedContextDisagg()->fap_context;
+    uint64_t region_id = 1;
+
+    EngineStoreServerWrap server = {
+        .tmt = &global_context.getTMTContext(),
+        .proxy_helper = proxy_helper.get(),
+    };
+
+    kvstore->getStore().store_id.store(1, std::memory_order_release);
+    kvstore->debugMutStoreMeta().set_id(1);
+    ASSERT_EQ(1, kvstore->getStoreID());
+    ASSERT_EQ(1, kvstore->clonedStoreMeta().id());
+    FailPointHelper::enableFailPoint(FailPoints::force_fap_worker_throw);
+    FailPointHelper::enableFailPoint(FailPoints::force_set_fap_candidate_store_id);
+    // The FAP will fail because it doesn't contain the new peer in region meta.
+    FastAddPeer(&server, region_id, 2333);
+    eventuallyPredicate(
+        [&]() { return fap_context->tasks_trace->queryState(region_id) == FAPAsyncTasks::TaskState::NotScheduled; });
```

Comment:
Seems this test does not go through the logic of `prev_state == FAPAsyncTasks::TaskState::Finished`?

https://github.com/pingcap/tiflash/blob/4bfa40d03176346822dede76ef6f9c2d62e05b53/dbms/src/Storages/KVStore/MultiRaft/Disagg/FastAddPeerContext.cpp#L247-L270